### DISCUSSION TITLE
Stage I - Volleyball Detection and Tracking

### DISCUSSION THREAD

**Discussion by shukkkur** [shukkkur/VolleyVision#5] [Show and tell]
## Welcome! 

By opening this discussion, I invite you 
- to ask questions about this stage of the project
- provide constructive criticism, 
- suggest new or improved methods.

 Your valuable input will help further enhance the accuracy and effectiveness of the tracking system. It will also help me and other users of VolleyVision to learn and grow! 

<h3 align="center"> <strong>Let's collaborate and explore the possibilities of advancing this project together!</strong></h3>

**Comment by shukkkur** [comment]
## Due to RoboFlow change of policy, now the number of hosted inference is limited.

**What it means?**
- When using the `roboflow` model you will have to adjust the code, so instead of public API key, you need to insert your own API key. Because the pubilic API key is most probably reached its limit (1000 per month)

![roboflow](https://github.com/shukkkur/VolleyVision/assets/78250180/df3d9fd5-a40d-4ca5-b953-aecda366fe18)

**Which lines of code to change?**
- if you are using `Stage I - Volleyball/detect.py`, change [line 158-160](https://github.com/shukkkur/VolleyVision/blob/445e768b7be2dab31d6f909bcc5bcfc2aeda34d9/Stage%20I%20-%20Volleyball/detect.py#L158-L160) with the following
![image](https://github.com/shukkkur/VolleyVision/assets/78250180/4fad5490-05e4-4207-8906-1de7c498136f)

- if you are using `Stage I - Volleyball/volley_track.py` change [line 138](https://github.com/shukkkur/VolleyVision/blob/7b894af12504b8b22defa4b111fe453089a7895d/Stage%20I%20-%20Volleyball/volley_track.py#L133)
- if you are using `Stage III - Court Detection/court_detect.py` change [line 77](https://github.com/shukkkur/VolleyVision/blob/7b894af12504b8b22defa4b111fe453089a7895d/Stage%20III%20-%20Court%20Detection/court_detect.py#L77)

**Where to get the API Key?**
1. Go to the [RoboFlow Model's Page](https://universe.roboflow.com/shukur-sabzaliev1/volleyball_v2/model/2)
![image](https://github.com/shukkkur/VolleyVision/assets/78250180/18c10b3a-4280-4eaa-bdf8-38614a7edc97)
2. Sign in using any of the Methods, reload the page and scroll down
![image](https://github.com/shukkkur/VolleyVision/assets/78250180/ae481187-2cb0-4309-9db8-05b4cc872719)

Not sure what is the inference limit for this API, so best of luck) 
Feel free to contact me)))

[Updated]

**Reply by BreeZe6167** [reply]
Hi, I'm trying to detect ball with your project now.
But it seems like the roboflow page was closed or delete?
How can I do?
(Thanks and sorry for my bad English.
![image](https://github.com/shukkkur/VolleyVision/assets/151605066/12769545-4e33-4059-92e1-6b00ec9fc228)

**Reply by shukkkur** [reply]
@BreeZe6167 Hi! Thanks for using the discussion page. 
I was cleaning the old volleyball dataset that consisted of 25k images. Mainly removed duplicates
Here - [19k_version](https://universe.roboflow.com/shukur-sabzaliev1/volleyball_v2/browse?queryText=&pageSize=50&startingIndex=0&browseQuery=true)

**Reply by BreeZe6167** [reply]
Thanks!

**Comment by Duchoang1123** [comment]
Error reading video file
loading Roboflow workspace...
loading Roboflow project...
Traceback (most recent call last):
  File "/content/VolleyVision/Stage I - Volleyball/detect.py", line 166, in <module>
    model = project.version(18).model
  File "/usr/local/lib/python3.10/dist-packages/roboflow/core/project.py", line 336, in version
    raise RuntimeError("Version number {} is not found.".format(version_number))
RuntimeError: Version number 18 is not found.

I tried to run and got this error?
I already pasted the api as recommended

**Reply by shukkkur** [reply]
Hi @Duchoang1123 

Deleting the old dataset (25k) resulted in deleting the model 😢 Didn't properly consider that. But hey, use [this version](https://universe.roboflow.com/shukur-sabzaliev1/volleyball_v2/model/2). 

Maybe later I will share YOLOv8 version I trained on 19k version dataset (1280x1280 image size). 
I got 90% precision and 80% recall.

**Reply by Duchoang1123** [reply]
Thank you @shukkkur !

I've tried it, the model works very accurately! Keep up the great work!

**Reply by BlackStarKos** [reply]
When you say to use the version above, do you mean the entire website itself? i assumed there was another repo or something where i could paste the new version number into the code provided.

**Reply by shukkkur** [reply]
@KosiReq oh I see. Scroll down to **Code Snippets** section. There you will see your API key. 
![image](https://github.com/shukkkur/VolleyVision/assets/78250180/f59048d6-818b-40c4-b820-a288a41d523d)
and then change [line 158](https://github.com/shukkkur/VolleyVision/blob/445e768b7be2dab31d6f909bcc5bcfc2aeda34d9/Stage%20I%20-%20Volleyball/detect.py#L158)

**Reply by BlackStarKos** [reply]
So I got to this same error when trying to run the code on my machine after changing the api key. 

<img width="854" alt="Screenshot 2024-07-10 000849" src="https://github.com/shukkkur/VolleyVision/assets/54788019/fa5a19a0-35f2-483b-a52d-fc9bdb5f72c0">

**Reply by shukkkur** [reply]
@KosiReq 
![image](https://github.com/shukkkur/VolleyVision/assets/78250180/2489e820-74bf-4b59-827d-44990f886ddf)

Insert these three lines to the code

**Comment by 1xyh1** [comment]
https://universe.roboflow.com/shukur-sabzaliev1/volleyball_v2/browse?queryText=&pageSize=50&startingIndex=0&browseQuery=true
https://universe.roboflow.com/shukur-sabzaliev1/volleyball_v2/model/2
Hello author, I would like to ask if the links you mentioned above have been deleted. When I log in, they all show "does not exist" like the first floor. If they have not been deleted, please let me know so that I can determine if it is my own network problem.Thanks!

**Reply by shukkkur** [reply]
The links you provided are accessible for me. So, probably yes your network.

https://universe.roboflow.com/shukur-sabzaliev1/volleyball_v2

**Reply by 1xyh1** [reply]
Thanks,I 'll find some way to improve my network.

**Comment by Phat-dev03704** [comment]
thank you i am developing fault detection in volleyball thanks to your platform
